tests/test_quantum_info_operations.py

```diff
@@ -109,3 +111,16 @@ def test_partial_trace(backend, density_matrix):
     Id = backend.identity_density_matrix(1, normalize=True)
 
     backend.assert_allclose(traced, Id)
+
+
+def test_matrix_power(backend):
+    nqubits, power = 2, 2
+    dims = 2**nqubits
+
+    state = random_density_matrix(dims, backend=backend)
+
+    power = matrix_power(state, power, backend)
+
+    backend.assert_allclose(
+        float(backend.np.real(backend.np.trace(power))), purity(state, backend=backend)
+    )
```

Comment:
```suggestion
@pytest.mark.parametrize(power, [2, 2., "2"])
def test_matrix_power(backend, power):
    nqubits = 2
    dims = 2**nqubits

    state = random_density_matrix(dims, backend=backend)

    if isinstance(power, str):
        with pytest.raises(TypeError):
            return matrix_power(state, power, backend)

    power = matrix_power(state, power, backend)

    backend.assert_allclose(
        float(backend.np.real(backend.np.trace(power))), purity(state, backend=backend)
    )
```
this should fix the coverage, if we want to keep the check